packages/core/elasticsearch/core-elasticsearch-server-internal/src/elasticsearch_service.ts

```diff
@@ -155,11 +160,21 @@ export class ElasticsearchService
             'Refer to https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-scripting-security.html for more info.'
         );
       }
+
+      capabilities = getElasticsearchCapabilities({
+        clusterInfo: await firstValueFrom(this.clusterInfo$!),
+      });
+    } else {
+      // skipStartupConnectionCheck is only used for unit testing, we default to base capabilities
+      capabilities = {
+        serverless: false,
+      };
     }
 
     return {
       client: this.client!,
       createClient: (type, clientConfig) => this.createClusterClient(type, config, clientConfig),
+      getCapabilities: () => capabilities,
```

Comment:
So, technically, capabilities are bound to a given cluster, so a given cluster. However attaching the API to `ClusterClient` was way more complex, and made the caching way harder too.

I doubt we're connecting Kibana to multiple cluster, especially mixed serverless/non-serverless ones anytime soon, so I just KISS.